packages/@aws-cdk/aws-stepfunctions-tasks/test/emr-add-step.test.ts

```diff
@@ -0,0 +1,168 @@
+import sfn = require('@aws-cdk/aws-stepfunctions');
+import cdk = require('@aws-cdk/core');
+import tasks = require('../lib');
+
+let stack: cdk.Stack;
+
+beforeEach(() => {
+  // GIVEN
+  stack = new cdk.Stack();
+});
+
+test('Add Step with static ClusterId and StepConfiguration', () => {
+  // WHEN
+  const task = new sfn.Task(stack, 'Task', {
+      task: new tasks.EmrAddStep({
+        clusterId: 'ClusterId',
+        stepConfiguration: sfn.TaskInput.fromObject({
+          Name: 'StepName'
+        }),
+        integrationPattern: sfn.ServiceIntegrationPattern.SYNC
+      })
+    });
+
+  // THEN
+  expect(stack.resolve(task.toStateJson())).toEqual({
+    Type: 'Task',
+    Resource: {
+      'Fn::Join': [
+        '',
+        [
+          'arn:',
+          {
+            Ref: 'AWS::Partition',
+          },
+          ':states:::elasticmapreduce:addStep.sync',
+        ],
+      ],
+    },
+    End: true,
+    Parameters: {
+      ClusterId: 'ClusterId',
+      Step: {
+        Name: 'StepName'
+      }
+    },
+  });
+});
+
+test('Terminate cluster with ClusterId from payload and static StepConfiguration', () => {
+  // WHEN
+  const task = new sfn.Task(stack, 'Task', {
+      task: new tasks.EmrAddStep({
+        clusterId: sfn.TaskInput.fromDataAt('$.ClusterId').value,
```

Comment:
This can be `sfn.Data.stringAt('$.ClusterId')` I believe, which is slightly simpler.